cmd/zoekt-sourcegraph-indexserver/cleanup.go

```diff
@@ -98,19 +109,25 @@ func cleanup(indexDir string, repos []uint32, now time.Time, shardMerging bool)
 	}
 
 	// index: Move missing repos from trash into index
+	// index: Restore deleted or tombstoned repos.
 	for _, repo := range repos {
 		// Delete from index so that index will only contain shards to be
 		// trashed.
 		delete(index, repo)
 
-		shards, ok := trash[repo]
-		if !ok {
-			continue
+		if shards, ok := trash[repo]; ok {
+			log.Printf("restoring shards from trash for %v", repo)
+			moveAll(indexDir, shards)
+			shardsLog(indexDir, "restore", shards)
 		}
 
-		log.Printf("restoring shards from trash for %v", repo)
-		moveAll(indexDir, shards)
-		shardsLog(indexDir, "restore", shards)
+		if shardMerging {
+			if s, ok := tombtones[repo]; ok {
+				log.Printf("removing tombstone for %v", repo)
+				zoekt.UnsetTombstone(s.Path, repo)
```

Comment:
you are ignoring the error here. I don't think we should fail, but we should atleast log?

Reply:
Oh yes, good find. That was not intentional. We shouldn't be writting to the shards log either.